selfdrive/ui/qt/offroad/wifiManager.cc

```diff
@@ -243,63 +164,32 @@ QVector<QDBusObjectPath> WifiManager::get_active_connections() {
 }
 
 bool WifiManager::isKnownConnection(const QString &ssid) {
-  return !getConnectionPath(ssid).path().isEmpty();
+  return (bool)getConnectionPath(ssid);
 }
 
 void WifiManager::forgetConnection(const QString &ssid) {
-  const QDBusObjectPath &path = getConnectionPath(ssid);
-  if (!path.path().isEmpty()) {
-    QDBusInterface nm2(NM_DBUS_SERVICE, path.path(), NM_DBUS_INTERFACE_SETTINGS_CONNECTION, bus);
-    nm2.call("Delete");
+  if (auto path = getConnectionPath(ssid)) {
```

Comment:
don't do assignment in conditions

Reply:
think it's more concise and clear to check std::optional variable in this way. How about keeping it as is? Otherwise we have to change many places.

Reply:
I guess this is alright for now, but I think std::optional is super confusing. Just reading `auto path = getConnectionPath(ssid)`, I have no idea what type `path` is.